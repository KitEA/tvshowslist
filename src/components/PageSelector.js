import React from "react";
import PropTypes from 'prop-types';

const PageSelector = ({ previousPage, nextPage }) => {
    return (
        <div className="page-selector">
            <button className="previous-page-button" onClick={previousPage}>&#8592; Previous page</button>
            <button className="next-page-button" onClick={nextPage}>Next page &#8594;</button>
        </div>
    )
}

PageSelector.propTypes = {
    previousPage: PropTypes.func.isRequired,
    nextPage: PropTypes.func.isRequired
}

export default PageSelector;