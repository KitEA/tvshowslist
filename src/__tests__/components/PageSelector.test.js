import React from "react";
import renderer from "react-test-renderer";
import PageSelector from "../../components/PageSelector";

describe("PageSelector", () => {
  it("matches the snapshot", () => {
    const tree = renderer
      .create(<PageSelector previousPage={() => {}} nextPage={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
