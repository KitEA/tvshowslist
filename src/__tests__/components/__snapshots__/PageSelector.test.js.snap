// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSelector matches the snapshot 1`] = `
<div
  className="page-selector"
>
  <button
    className="previous-page-button"
    onClick={[Function]}
  >
    ← Previous page
  </button>
  <button
    className="next-page-button"
    onClick={[Function]}
  >
    Next page →
  </button>
</div>
`;
